File: internal/model/analysis.go
package model

import (
	"time"
)

// Analysis 分析历史模型
type Analysis struct {
	ID             string     `json:"id" gorm:"primaryKey"`
	SessionID      string     `json:"session_id" gorm:"index"`              // 会话ID（可选）
	ResultsData    string     `json:"results_data" gorm:"type:text"`        // 执行结果数据（JSON格式）
	AnalysisResult string     `json:"analysis_result" gorm:"type:text"`     // 分析结果
	Question       string     `json:"question" gorm:"type:text"`            // 用户提问（可选）
	ToolCalls      []ToolCall `json:"tool_calls" gorm:"serializer:json"`    // 工具调用记录
	CreatedAt      time.Time  `json:"created_at"`
}

// SemanticInsight 语义化解读结果
type SemanticInsight struct {
	Summary        string   `json:"summary"`                    // 简短摘要
	RiskLevel      string   `json:"risk_level"`                 // normal | warning | critical
	Trend          string   `json:"trend,omitempty"`            // 趋势分析
	Recommendation string   `json:"recommendation,omitempty"`   // 建议操作
	KeyMetrics     []Metric `json:"key_metrics,omitempty"`      // 关键指标
}

// Metric 关键指标
type Metric struct {
	Name      string  `json:"name"`
	Value     string  `json:"value"`
	Status    string  `json:"status"`    // normal | warning | critical
	Threshold string  `json:"threshold,omitempty"`
}

// ContextualAction 上下文相关的推荐操作
type ContextualAction struct {
	ID            string  `json:"id"`
	Label         string  `json:"label"`
	Icon          string  `json:"icon"`
	Command       string  `json:"command,omitempty"`
	Highlight     bool    `json:"highlight"`
	AIRecommended bool    `json:"ai_recommended"`
	Confidence    float64 `json:"confidence,omitempty"`
	RiskLevel     string  `json:"risk_level"`
	Description   string  `json:"description,omitempty"`
}

// HistoricalCorrelation 历史关联分析
type HistoricalCorrelation struct {
	HostID          string `json:"host_id"`
	CurrentIssue    string `json:"current_issue"`
	SimilarIncident string `json:"similar_incident,omitempty"`
	OccurredAt      string `json:"occurred_at,omitempty"`
	Resolution      string `json:"resolution,omitempty"`
	Confidence      float64 `json:"confidence"`
}

File: web/src/api/analysis.ts
import { request } from './request'

export interface AnalyzeRequest {
  results: unknown[]
  question?: string
}

export interface ToolCall {
  tool: string
  params: Record<string, unknown>
  result?: unknown
  error?: string
}

export interface AnalyzeResponse {
  id: string
  analysis_result: string
  question?: string
  tool_calls?: ToolCall[]
  created_at: string
}

export interface Analysis {
  id: string
  session_id?: string
  results_data: string
  analysis_result: string
  question?: string
  tool_calls?: ToolCall[]
  created_at: string
}

export interface AnalysisHistoryParams {
  session_id?: string
  limit?: number
}

// 语义化解读相关类型
export interface SemanticInsightRequest {
  tool_name: string
  tool_result: unknown
  host_id?: string
}

export interface Metric {
  name: string
  value: string
  status: 'normal' | 'warning' | 'critical'
  threshold?: string
}

export interface SemanticInsight {
  summary: string
  risk_level: 'normal' | 'warning' | 'critical'
  trend?: string
  recommendation?: string
  key_metrics?: Metric[]
}

// 上下文操作推荐相关类型
export interface ContextualAction {
  id: string
  label: string
  icon: string
  command?: string
  highlight: boolean
  ai_recommended: boolean
  confidence?: number
  risk_level: 'low' | 'medium' | 'high'
  description?: string
}

export interface SuggestActionsRequest {
  tool_results: Array<{
    tool_name: string
    result: unknown
    host_id?: string
  }>
}

// ============================================================================
// Analysis API Functions
// ============================================================================

export function analyzeResults(data: AnalyzeRequest): Promise<AnalyzeResponse> {
  return request.post<AnalyzeResponse>('/analysis/analyze', data)
}

export function getAnalysisHistory(params?: AnalysisHistoryParams): Promise<Analysis[]> {
  return request.get<Analysis[]>('/analysis/history', { params })
}

export function getAnalysis(id: string): Promise<Analysis> {
  return request.get<Analysis>(`/analysis/${id}`)
}

export function deleteAnalysis(id: string): Promise<void> {
  return request.delete(`/analysis/${id}`)
}

// 获取工具执行结果的语义化解读
export function getSemanticInsight(data: SemanticInsightRequest): Promise<SemanticInsight> {
  return request.post<SemanticInsight>('/analysis/semantic', data)
}

// 获取上下文相关的推荐操作
export function getSuggestedActions(data: SuggestActionsRequest): Promise<ContextualAction[]> {
  return request.post<ContextualAction[]>('/analysis/suggest-actions', data)
}

// 历史关联分析相关类型
export interface HistoricalCorrelationRequest {
  host_id: string
  current_issue: string
  issue_type?: string
  keywords?: string[]
}

export interface HistoricalCorrelation {
  host_id: string
  current_issue: string
  similar_incident?: string
  occurred_at?: string
  resolution?: string
  confidence: number
}

// 获取历史关联分析
export function getHistoricalCorrelation(data: HistoricalCorrelationRequest): Promise<HistoricalCorrelation> {
  return request.post<HistoricalCorrelation>('/analysis/correlation', data)
}






